### PR TITLE
Fixes #4265 - Clarify React.render note in documentation

### LINKED ISSUE #4265: Documentation of React.render seems to self-contradict
Correct me if I misunderstood something, but the documentation of the React.render method says:

> If the ReactElement was previously rendered into container, this will perform an update on it and only mutate the DOM as necessary[...]

But later...

> *Note*: React.render() replaces the contents of the container node you pass in.[...]

In practice, what seems to happen corresponds with the *Note*

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
This updates the note text to that found in #4265 in an effort to clarify`React.render()`'s documentation.

> Note:
> 
> `React.render()` controls the contents of the container node you pass in*. Any existing DOM elements
> inside are replaced when first called. Later calls use React’s DOM diffing algorithm for efficient 
> updates.
>
> \* `React.render()` does not modify the container node (only modifies the children of the container). In
> the future, it may be possible to insert a component to an existing DOM node without overwriting 
> the existing children.

Unsure if you want the function call highlight in the second paragraph.

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
@jimfb fixed the merge target

**Comment by jimfb** [comment]
@KevinTCoughlin Thanks!  I'll let this sit for a couple days so people have a chance to see it, and if there are no objections, we'll merge.

File: docs/docs/ref-01-top-level-api.md [hunk 1]
```diff
@@ -88,10 +88,14 @@ If the ReactElement was previously rendered into `container`, this will perform
...
 > Note:
+> 
+> `React.render()` controls the contents of the container node you pass in*. Any existing DOM elements
+> inside are replaced when first called. Later calls use React’s DOM diffing algorithm for efficient 
+> updates.
 >
-> `React.render()` replaces the contents of the container node you
-> pass in. In the future, it may be possible to insert a component to an
-> existing DOM node without overwriting the existing children.
+> \* `React.render()` does not modify the container node (only modifies the children of the container). In
```
Comment (zpao) [inline docs/docs/ref-01-top-level-api.md]: Seems fine but can you remove the `\*` bit?

File: docs/docs/ref-01-top-level-api.md [hunk 2]
```diff
@@ -89,9 +89,13 @@ If the optional callback is provided, it will be executed after the component is
 
 > Note:
 >
-> `React.render()` replaces the contents of the container node you
-> pass in. In the future, it may be possible to insert a component to an
-> existing DOM node without overwriting the existing children.
+> `React.render()` controls the contents of the container node you pass in*. Any existing DOM elements
```
Comment (cody) [inline docs/docs/ref-01-top-level-api.md]: Really nice. Just the superfluous * should be removed.

Comment (jimfb) [inline docs/docs/ref-01-top-level-api.md]: The star is there to indicate a reference to the footnote/clarification below.  It's a way of connecting the additional information to that sentence without breaking the train of thought there.

Comment (KevinTCoughlin) [inline docs/docs/ref-01-top-level-api.md]: Yea, the `\*` removed in 60a081e66b634a269caf76549b78489dad74f9c9 was there to escape the asterisk for markdown formatting and reference the footnote below.

Comment (zpao) [inline docs/docs/ref-01-top-level-api.md]: Ohh, I see. I think it's better to leave out the footnote demarkation. The content reads well flowing into the next block without it.

Comment (KevinTCoughlin) [inline docs/docs/ref-01-top-level-api.md]: @zpao - sounds good, removed the footnote demarcation in ee11412

**Comment by jimfb** [comment]
Looks good.  Thanks @KevinTCoughlin!

**Pull request merged**